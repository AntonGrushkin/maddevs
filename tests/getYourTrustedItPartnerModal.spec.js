import {
  mount
} from '@vue/test-utils';
import getYourTrustedItPartnerModal from '@/components/Modals/get-your-trusted-it-partner-modal';

describe('Get your trusted it partner modal', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = mount(getYourTrustedItPartnerModal, {
      stubs: ['ValidationProvider', 'ValidationObserver', 'modal', 'perfect-scrollbar']
    });
  });

  // ------ IMPORTANT ----- //
  test('is a Vue instance', () => {
    expect(wrapper.exists()).toBeTruthy();
  });
  
  test('renders correctly', () => {
    expect(wrapper.element).toMatchSnapshot();
  });
  // --------------------- //

  test('sets the correct default data', () => {
    expect(typeof getYourTrustedItPartnerModal.data).toBe('function');
    const defaultData = getYourTrustedItPartnerModal.data();
    expect(
      defaultData.agreeWithPrivacyPolicy &&
      defaultData.agreeToGetMadDevsDiscountOffers
    ).toEqual(false);
    expect(defaultData.inputId).toEqual('get-your-trusted-it-partner');
  });

  test('has a functions', () => {
    expect(
      typeof getYourTrustedItPartnerModal.methods.getPrivacyCheckboxState && 
      typeof getYourTrustedItPartnerModal.methods.getDiscountOffersCheckboxState &&
      typeof getYourTrustedItPartnerModal.methods.autosize
    ).toBe('function');
  });

  test('call functions with params and change variables state', () => {
    wrapper.vm.getPrivacyCheckboxState(true);
    wrapper.vm.getDiscountOffersCheckboxState(true);

    expect(
      wrapper.vm.$data.agreeWithPrivacyPolicy &&
      wrapper.vm.$data.agreeToGetMadDevsDiscountOffers
    ).toEqual(true);
  });

  test('autosize function should add value in event key', () => {
    const event = {
      target: {
        style: {
          height: ''
        },
        scrollHeight: 100
      }
    };

    wrapper.vm.autosize(event);
    expect(event.target.style.height).toEqual('100px');
  });
});
