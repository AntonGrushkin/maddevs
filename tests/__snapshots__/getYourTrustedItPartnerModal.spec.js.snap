// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Get your trusted it partner modal renders correctly 1`] = `
<modal-stub
  classes="modal_container"
  height="auto"
  name="get-your-trusted-it-partner"
>
  <button
    class="close-modal"
  >
    <img
      alt="Close modal"
      src="@/assets/img/common/close-icon.svg"
    />
  </button>
   
  <perfect-scrollbar-stub
    class="modal_scrollbar custom-scrollbar safari_only"
    options="[object Object]"
  >
    <validationobserver-stub />
  </perfect-scrollbar-stub>
</modal-stub>
`;
